pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -513,7 +513,13 @@ func (p *Provisioner) setEndpoint(sc *storagev1.StorageClass) {
 
 func (p *Provisioner) setRegion(sc *storagev1.StorageClass) {
 	const key = "region"
-	p.region = sc.Parameters[key]
+	if len(sc.Parameters[key]) > 0 {
+		p.region = sc.Parameters[key]
+	} else {
+		// If user does not define, then set region to Zonegroup, since RGW internally maps
+		// aws region to ZoneGroup
+		p.region = p.objectContext.ZoneGroup
+	}
```

Comment:
I'm a little bit confused. What is the point of overriding `region` if the default is mapped to `ZoneGroup` and if the user has to create a ZoneGroup resource if they want to override region? Would it be simpler to just remove this parameter and rely on users to create a `ZoneGroup` instead?

Reply:
@BlaineEXE : `region` is an optional value for storage-class. If it is not defined then we need to fill up with `Zonegroup` from object store context, This value changes in different scenarios: 
1.) normal cluster, it is name of object store
2.) external cluster it is `default`
3.) if multiple zones are configured then it depends on value set in object-store CR.

If the user wants to use a custom region then he defines `Zonegroup` that's different scenario. Hope this avoids the confusion

Reply:
In all three of those cases, I don't think we need to have the `region` parameter.
1. If external cluster with no zone group defined, use `default`
2. If external cluster with external zone group defined, use the zone group
3. If internal cluster with no zone group defined, use name of object store
4. If internal cluster with zone group defined, use the zone group

Is this not correct? Are there any cases missing?

Reply:
@BlaineEXE : Please see https://github.com/rook/rook/pull/9754#discussion_r822287813 why the region is needed.
Above cases are right, but please note 2.) is not support in Rook atm